Remove "wprintf" from PAL

AaronRobinsonMSFT opened the pull request:
The last usage was in `FormatMessageW`, which
only used wprintf syntax for messages that used
embedded printf format between two "!". We do
not use this feature in any of our messages so
this code path is technically unused.

Remove `_snwprintf_s` from PAL

Remove `_vsnwprintf_s` from PAL

`_woutput_s` was not removed because it is intertwined
with the implementation for printf. The linker will just
prune the implementation anyways.

Remove all wprintf testing from PAL

AaronRobinsonMSFT committed:
Remove final usage of *wprintf* from PAL

The last usage was in FormatMessageW, which
only used wprintf syntax for messages that used
embeded printf format between two "!". We do
not use this feature in any of our messages so
this code path is technically unused.

AaronRobinsonMSFT committed:
Remove _snwprintf_s from PAL

AaronRobinsonMSFT committed:
Remove _vsnwprintf_s from PAL

_woutput_s was not removed because it is intertwined
with the implementation for printf. The linker will just
prune the implementation anyways.

Remove all wprintf testing from PAL

AaronRobinsonMSFT commented:
/cc @am11 @jkotas @davidwrighton

am11 commented on src/coreclr/pal/tests/palsuite/CMakeLists.txt:
```diff
@@ -322,24 +322,6 @@ add_executable_clr(paltests
   c_runtime/vsprintf/test7/test7.cpp
   c_runtime/vsprintf/test8/test8.cpp
   c_runtime/vsprintf/test9/test9.cpp
-  c_runtime/vswprintf/test1/test1.cpp
```
These are also referenced in `compilableTests.txt` `paltestlist.txt` and `paltestlist_to_be_reviewed.txt`.

Overall in PR branch, it looks like these files needs reviewing: 
```sh
$ git grep -l snwprintf_s 
src/coreclr/pal/tests/palsuite/compilableTests.txt
src/coreclr/pal/tests/palsuite/paltestlist.txt
src/coreclr/pal/tests/palsuite/paltestlist_to_be_reviewed.txt
src/coreclr/vm/clrtocomcall.cpp
src/coreclr/vm/commtmemberinfomap.cpp
src/coreclr/vm/dispatchinfo.cpp
src/coreclr/vm/dwbucketmanager.hpp
src/coreclr/vm/dwreport.cpp
src/native/corehost/hostmisc/pal.h
src/tests/Interop/COM/NativeServer/DispatchTesting.h
```

AaronRobinsonMSFT replied:
I'll update the txt files. The others are all conditioned to build under Windows, which is why this is okay.

am11 reviewed (approved):
Looks great! 👍

davidwrighton commented on src/coreclr/pal/src/misc/fmtmessage.cpp:
```diff
@@ -479,115 +383,27 @@ FormatMessageW(
                     nCount = 0;
                     goto exit;
                 }
-                if ( *lpSourceString == '!' )
```
I'd like to see this fail in some way instead of silently not behaving like the windows FormatMessage api

AaronRobinsonMSFT replied:
I thought about that. Wasn't sure of if a fail fast or assert or something else was appropriate. Thoughts?

AaronRobinsonMSFT replied:
I'll error out like the other places in the function.

AaronRobinsonMSFT committed:
Remove mention of "wprintf" in PAL test files
FormatMessageW errors if embedded formating is used
Update PAL tests for FormatMessageW

janvorli reviewed (approved):
LGTM, thank you!

AaronRobinsonMSFT commented:
Issues are known and properly tagged above.
